packages/ngtools/webpack/src/transformers/find_image_domains_spec.ts

```diff
@@ -0,0 +1,153 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { tags } from '@angular-devkit/core';
+import * as ts from 'typescript';
+import { findImageDomains } from './find_image_domains';
+import { createTypescriptContext, transformTypescript } from './spec_helpers';
+
+function findDomains(
+  input: string,
+  importHelpers = true,
+  module: ts.ModuleKind = ts.ModuleKind.ES2020,
+) {
+  const { program, compilerHost } = createTypescriptContext(input, undefined, undefined, {
+    importHelpers,
+    module,
+  });
+  const domains = new Set<string>();
+  const getTypeChecker = () => program.getTypeChecker();
+  const transformer = findImageDomains(getTypeChecker, domains);
+
+  transformTypescript(input, [transformer], program, compilerHost);
+
+  return domains;
+}
+
+function inputTemplate(provider: string) {
+  /* eslint-disable max-len */
+  return tags.stripIndent`
+    export class AppModule {
+        static ɵfac = function AppModule_Factory(t) { return new (t || AppModule)(); };
+        static ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: AppModule, bootstrap: [AppComponent] });
+        static ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ providers: [
+                ${provider}
+            ], imports: [BrowserModule] });
+    }
+    (function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(AppModule, [{
+            type: NgModule,
+            args: [{
+                    declarations: [
+                        AppComponent
+                    ],
+                    imports: [
+                        BrowserModule,
+                        NgOptimizedImage
+                    ],
+                    providers: [
+                        ${provider}
+                    ],
+                    bootstrap: [AppComponent]
+                }]
+        }], null, null); })();
+    (function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(AppModule, { declarations: [AppComponent], imports: [BrowserModule,
+            NgOptimizedImage] }); })();
+    `;
+}
+
+describe('@ngtools/webpack transformers', () => {
+  describe('find_image_domains', () => {
+    it('should find a domain when a built-in loader is used with a string-literal-like argument', () => {
+      // Intentionally inconsistent use of quote styles in this data structure:
+      const builtInLoaders: Array<[string, string]> = [
+        ['provideCloudflareLoader("www.cloudflareUrl.com")', 'www.cloudflareUrl.com'],
```

Comment:
Super nit: the capitalization of "Url" in the URL feels a bit odd - lower case it?

Reply:
Yeah camelcase for URLs looks weird. Changed 'em